### PR TITLE
chore: upgrade Rust to 1.59

### PR DESCRIPTION
**Pull request by crepererum**
Mostly a few new clippy crates around `flat_map`, `and_then`, and
"underscore locks" (!!!):
https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_lock

### CODE REVIEW

File: db/src/lib.rs [hunk 1]
```diff
@@ -3023,7 +3022,7 @@ mod tests {
 
         let task = tokio::spawn(async move {
             sender.send(()).unwrap();
-            let _ = chunk_a.read();
+            let _read = chunk_a.read();
```
Comment (alamb) [inline db/src/lib.rs:3025]: this one seems unnecessary 😆

**Review by alamb — APPROVED** [review]
All in all I like the improvements to the code from the new clippy lints (which I haven't found to be the case all that often) 👍

File: db/src/lib.rs (hunk 1 above)
Comment (crepererum) [reply db/src/lib.rs:3025]: Yeah, you could also just drop get the lock and instantly drop it. I kept the underscore version for the sake of visibility (i.e. "a guard was created") and consistency.

Comment (alamb) [reply db/src/lib.rs:3025]: makes sense 👍

**Pull request merged**